### ISSUE TITLE
Art of Chording lessons

### ISSUE THREAD

**Issue by TilBlechschmidt** [didoesdigital/typey-type-data#4]
Hi!
I've started with steno a couple of weeks ago and am using Art of Chording for theory lessons. However, I prefer TypeyType over other sites when it comes to practicing. StenoGrind by Noah contains a word list and some sentences built from the lessons in the textbook. I invested some effort to convert them into custom lessons for TypeType. Would you be interested in include them directly?

While starting out I felt that while TypeyType is a great tool, it lacks dedicated learning material that coincides with the lessons included. There is a very short explanation for each one and they kinda work but definitely do not replace a full textbook. This is the reason why I'd love to see more lessons included that match with books that are already out there.

**Comment by TilBlechschmidt** [comment]
I've currently gathered data for the first sixteen lessons, though there are more (the latest database is not on GitHub so I only converted the ones available so far). ZIP file is attached.

[Art of Chording lessons (TypeyType).zip](https://github.com/didoesdigital/typey-type-data/files/7877446/Art.of.Chording.lessons.TypeyType.zip)

**Comment by didoesdigital** [comment]
Hi! Welcome to steno! Thanks for sharing your idea as well as investing in the effort to prepare custom lessons 🙏 

There seems to be a fair bit of interest for lessons more closely matching Art of Chording so it might be good to help support that somehow.

One option could be to update the fundamental lessons to make sure they cover the relevant words.

Another is to specifically create a "collection" of Art of Chording lessons and take care to adhere to the licensing used ((CC BY-SA 4.0)[https://github.com/morinted/art_of_chording/blob/main/LICENSE.txt]) e.g. proper attribution and not applying additional restrictions, which could be documented here: https://github.com/didoesdigital/typey-type-data#licenses

For the second option, I expect we'll also need to think about maintenance/how to keep the word lists up to date. What do you mean about the latest database is not on GitHub?

**Comment by TilBlechschmidt** [comment]
I am not quite sure which of the two options is best and how to decide. I might be mistaken but to me (as a learner) it seems like TypeyTypes early lessons cover different aspects of theory (or at least in a different order) when compared to Art of Chording. So merging the two might involve a bunch of refactoring which might not even be desirable.

Regarding the licensing I am not an expert but maybe it would suffice if we say "these lessons are covered by this license and attribution for content goes to Ted".

Concerning the database question: As a quick hack (to have some custom lessons I could use), I took the SQLite database behind StenoGrind by NoahRR (see https://github.com/NoahRR/The-Steno-Grind) and simply extracted his word lists. However, upon closer inspection, it seems like it is mostly comprised of the words, briefs, and sentences presented at the end of each AoC lesson. Assuming that the lesson structure stays the same, this could be the way to maintaining the lists. Extract the words, briefs, and sentences directly from AoC into say a YAML file (for easy updating), and compile those ever so often into the lesson format TypeyType expects.

**Comment by didoesdigital** [comment]
Good point on the ordering of material, Typey Type's early lessons (based roughly on [Learn Plover](https://sites.google.com/site/learnplover/home)) is probably too different to easily reconcile.

Grabbing the words from the end of the Art of Chording pages and popping into something to convert to Typey Type lesson format seems like a decent compromise for keeping them in sync. That'll require a bit of work on my end but I'll add it to the list.

In the meantime, what do you think about something like a [dedicated "custom lesson" spreadsheet](https://docs.google.com/spreadsheets/d/1JfLxv_GK6M_OpvtbVuPhvohCJ6He2R6d7_zpHTa8k20/edit?usp=sharing) prominently linked from Typey Type? e.g. 

<img width="1434" alt="image" src="https://user-images.githubusercontent.com/2476974/150908155-230853dd-452c-409e-93f1-3507a2ac03e7.png">

**Comment by TilBlechschmidt** [comment]
**Data entry:**
If we have a format and easy method of entering data, I could certainly help the effort by creating a couple of lessons from the lessons I am currently learning.

**Spreadsheets:**
I've come across the other community lesson spreadsheet before but did not want to add those 40 something sheets which would have made it a horrible experience to navigate. However, I do like the idea of having a dedicated spreadsheet for the AoC lessons! Here are some thoughts regarding the spreadsheets in general:

- `+` They are easy to understand and navigate for anyone
- `+` Barrier of contribution is relatively low with the suggestions functionality
- `-` Using them in Typey Type is a bit of a hassle
    - Navigating into the spreadsheet, finding the right sub-sheet, copying out just the right cells, pasting it into Typey Type are too many steps for my taste
- `-` I do not think there is a way to set "lesson default settings" for those, is there?
    - AoC lessons would probably benefit from having zero repetitions but an "infinite" word limit as the sentences themselves have repetition in them

Regarding the third bullet point, have you considered integrating the spreadsheets directly? AFAIK Google Sheets has an API which would make it possible to simply include the "spreadsheet lessons" into the clickable list of lessons (in a separate section) much like the OG lessons:
<img width="557" alt="image" src="https://user-images.githubusercontent.com/5037967/150976049-0ef29420-3606-4d6a-8c33-35bae5f97bce.png">
